create.ps1

```diff
@@ -1,274 +1,378 @@
+
+#region Account mapping
+$account = [PSCustomObject]$actionContext.Data
+
+# Convert isEnabled to boolean
+if (-not[String]::IsNullOrEmpty($account.isEnabled)) {
+    $account.isEnabled = [System.Convert]::ToBoolean($account.isEnabled)
+}
+
+# Add manager userGUID to account object
+# Note: this is only available after granting the account for the manager
+if ($true -eq $actionContext.Configuration.setManager) {
+    $account.managedByUserGUID = $actionContext.References.ManagerAccount.userGUID
+}
+else {
+    # If option to set manager isn't toggled, remove from account object
+    $account.PSObject.Properties.Remove("managedByUserGUID")
+}
+
+# AccountReference must have a value
+$outputContext.AccountReference = "Currently not available"
+#endregion Account mapping
+
+try {
+    # Validate correlation configuration
+    if ($actionContext.CorrelationConfiguration.Enabled) {
+        $correlationField = $actionContext.CorrelationConfiguration.accountField
+        $correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
+
+        if ([string]::IsNullOrEmpty($($correlationField))) {
+            throw "Correlation is enabled but not configured correctly"
+        }
+        if ([string]::IsNullOrEmpty($($correlationValue))) {
+            throw "Correlation is enabled but [accountFieldValue] is empty. Please make sure it is correctly mapped"
+        }
 
-# Troubleshooting
-# $account = [PSCustomObject]@{
-#     # Optional, if you use a sync to HelloID, make sure set the immutableId with the AD SID
-#     immutableId          = [string]$adUser.sid;
-#     userName             = "user@enyoi.onmicrosoft.com"
-#     firstName            = "John"
-#     lastName             = "Doe"
-#     contactEmail         = "user@enyoi.nl"
-#     isEnabled            = $false
-#     password             = "Tools4ever!"
-#     mustChangePassword   = $true
-#     managedByUserGUID    = "178cb594-09b1-4645-892a-9fe6a8eb8a08" # Only available after grant for manager
-#     # If you use a sync to HelloID, make sure to specify the same source name, e.g. 'enyoi.local'
-#     source               = $(Get-adforest).RootDomain
-#     userAttributes = @{
-#         EmployeeId           = "12345678"
-#         Department           = "Test"
-#         Title                = "Tester"
-#         PhoneNumber          = "+3167652102"
-#         SAMAccountName       = "Test"
-#     }
-# }
-# $dryRun = $false
-
-# Create or Correlate user
-try{
-    if(-Not($dryRun -eq $True)) {
         # Create authorization headers with HelloID API key
-        $pair = "${apiKey}:${apiSecret}"
-        $bytes = [System.Text.Encoding]::ASCII.GetBytes($pair)
-        $base64 = [System.Convert]::ToBase64String($bytes)
-        $key = "Basic $base64"
-        $headers = @{"authorization" = $Key}
-
-        # Define specific endpoint URI
-        if($PortalBaseUrl.EndsWith("/") -eq $false){
-            $PortalBaseUrl = $PortalBaseUrl + "/"
+        try {
```

Comment:
I'm not sure if this is possible. We now use a try/catch in a try/catch in my opion (only when realy needed). For example for making a authorization header, correlating account and creating account. Is it not better to filter the error in the general try/catch for example
if action = "authorization header" then this error logging
if action = "correlating account" then other error logging

Reply:
As each action should be a seperate and standalone code block, they each require their own try/catch.
The "global" try/catch/finally is there mainly for the finally, to perform specific actions always after each action.
So the sepereate try/catch(es) with one "gloval" try/catcch is by design.
Let's discuss this when possible.

Reply:
For now, we will keep it this way, as there is no right/wrong on this for now.
If it later proves to be better to use one single try/catch, we will change this.